### DISCUSSION TITLE
Write wrapper function around `aesara.scan` with nice, intuitive signature

### DISCUSSION THREAD

**Discussion by michaelosthege** [pymc-devs/pymc3#6190] [Ideas]
We can just take a bunch of kwargs with informative names and docstrings, and behind the scenes feed things into the less intuitive parts of the `aesara.scan` signature such as the `output_info` kwarg.

@ricardoV94

**Comment by twiecki** [comment]
Could it be a decorator for a function that's executed inside the loop?

```python
k = tt.iscalar("k")
A = tt.vector("A")

# Compute A**k
@aet.for_(k)
def loop(result=0):
    result += result * A
    return result
```
